src/Core/TileMesh.js

```diff
@@ -158,14 +158,13 @@ TileMesh.prototype.setTexturesLayer = function setTexturesLayer(textures, layerT
     }
 };
 
-TileMesh.prototype.getLayerTextures = function getLayerTextures(layerType, layerId) {
-    const mat = this.material;
-    return mat.getLayerTextures(layerType, layerId);
-};
-
-TileMesh.prototype.isColorLayerLoaded = function isColorLayerLoaded(layerId) {
+TileMesh.prototype.isColorLayerLoaded = function isColorLayerLoaded(layer) {
     const mat = this.material;
-    return mat.getColorLayerLevelById(layerId) > -1;
+    const textures = mat.getLayerTextures(layer);
+    if (textures.length) {
+        return textures[0].coords.zoom > -1;
```

Comment:
should not it be `textures.every(texture => texture.coords.zoom == this.level)` or `textures.every(texture => somefun(layer, this.extent, texture.coords)` for a suitable function ?

Reply:
Actually it doesn't matter, since we always set textures in one shot (this is what getColorLayerLevelById was doing anyway)